Data/ByteString.hs

```diff
@@ -1229,11 +1229,11 @@ group xs
 
 -- | The 'groupBy' function is the non-overloaded version of 'group'.
 groupBy :: (Word8 -> Word8 -> Bool) -> ByteString -> [ByteString]
-groupBy k xs
-    | null xs   = []
-    | otherwise = unsafeTake n xs : groupBy k (unsafeDrop n xs)
+groupBy k xs = case uncons xs of
+  Nothing     -> []
+  Just (h, t) -> take n xs : groupBy k (drop n xs)
```

Comment:
`unsafeTake` and `unsafeDrop` still make sense here, shaving off a couple of picoseconds.